main.nf

```diff
@@ -770,15 +853,60 @@ process get_software_versions {
 
     output:
     file "software_versions.csv"
+    file 'software_versions_mqc.yaml' into software_versions_yaml
 
     script:
     """
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
+    fastqc --version &> v_fastqc.txt
+    multiqc --version &> v_multiqc.txt
+    vsearch --version &> v_vsearch.txt
+    muscle -version &> v_muscle.txt
+    python -c "import presto; print(presto.__version__)" > v_presto.txt
+    python -c "import changeo; print(changeo.__version__)" > v_changeo.txt
+    echo \$(R --version 2>&1) > v_R.txt
+    Rscript -e "library(shazam); write(x=as.character(packageVersion('shazam')), file='v_shazam.txt')"
+    Rscript -e "library(alakazam); write(x=as.character(packageVersion('alakazam')), file='v_alakazam.txt')"
+    Rscript -e "library(tigger); write(x=as.character(packageVersion('tigger')), file='v_tigger.txt')"
     scrape_software_versions.py &> software_versions_mqc.yaml
     """
 }
 
+//MultiQC
+process multiqc {
+    publishDir "${params.outdir}/MultiQC", mode: 'copy'
+
+    input:
+    file multiqc_config from ch_multiqc_config
+    file (fastqc:'fastqc/*') from fastqc_results.collect().ifEmpty([])
+    file ('software_versions/*') from software_versions_yaml.collect()
+    file workflow_summary from create_workflow_summary(summary)
+
+    output:
+    file "*multiqc_report.html" into multiqc_report
+    file "*_data"
+    file "multiqc_plots"
+
+    script:
+    rtitle = custom_runName ? "--title \"$custom_runName\"" : ''
+    rfilename = custom_runName ? "--filename " + custom_runName.replaceAll('\\W','_').replaceAll('_+','_') + "_multiqc_report" : ''
+    """
+    multiqc . -f $rtitle $rfilename --config $multiqc_config \\
+        -m custom_content -m fastqc
+    """
+}
+
+//Useful functions
+
+// Return file if it exists
```

Comment:
This can go entirely... if you use `file(...)` for any input, you don't need this funciton anymore. There is a file(..., checkIfExists: true)` functionality available that does exactly that :+1: